components/Profile/PointsShowcase.jsx

```diff
@@ -0,0 +1,147 @@
+import {
+  useEffect, useState, useRef, useMemo,
+} from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import Link from 'next/link';
+import {
+  Card, Spin, Row, Col, Typography,
+} from 'antd';
+import { useScript } from './hooks';
+import { shuffleArray } from './utils';
+
+const { Paragraph, Title } = Typography;
+
+const TweetLoader = styled.div`
+  display: flex;
+  justify-content: center;
+  margin: 16px 16px 32px;
+`;
+const StyledCol = styled(Col)`
+  max-width: 290px;
+`;
+
+const StyledCard = styled(Card)`
+  max-width: 282px;
+`;
+
+const TweetEmbed = ({ points, tweetId }) => {
+  const tweetLoaded = useRef(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const status = useScript('https://platform.twitter.com/widgets.js');
+
+  useEffect(() => {
+    const handleTweetLoad = () => {
+      setIsLoading(false);
+      tweetLoaded.current = true;
+    };
+
+    const loadTweetEmbedded = () => {
+      if (status !== 'ready') return;
+      if (tweetLoaded.current) return;
+
+      window.twttr.widgets.createTweet(
+        tweetId,
+        document.getElementById(`tweet-container-${tweetId}`),
+        { width: 250, cards: 'hidden', conversation: 'none' },
+      );
+      tweetLoaded.current = true;
+    };
+
+    // Observe changes inside the tweet node,
+    // to update loading state in case of success or errors
+    const element = document.getElementById(`tweet-container-${tweetId}`);
+    const observer = new MutationObserver((mutationsList) => {
```

Comment:
even if [this](https://developer.twitter.com/en/docs/twitter-for-websites/embedded-tweets/guides/embedded-tweet-javascript-factory-function) says that `window.twttr.widgets.createTweet` returns a Promise - in case of a tweet not found, I don't fall to "catch" block and can't cancel loading state, that's why I use `MutationObserver`